lib/shim/promise-shim.js

```diff
@@ -362,13 +363,59 @@ function _wrapResolver(context, fn) {
 }
 
 /**
- * @param shim
- * @param fn
- * @param name
- * @param useAllParams
+ *
+ * @param {object} params object passed to wrapHandler
+ * @param {Function} params.handler to wrap
+ * @param {number} params.index index of argument
+ * @param {number} params.argsLength length of args
+ * @param {boolean} params.useAllParams flag to use all params
+ * @param {object} params.ctx context passed in to store the next handler and isWrapped
+ * @param {object} params.shim shim instance
+ * @returns {Function} wrapped function
+ */
+function wrapHandler({ handler, index, argsLength, useAllParams, ctx, shim }) {
+  if (
+    !shim.isFunction(handler) || // Not a function
+    shim.isWrapped(handler) || // Already wrapped
```

Comment:
I see that these comments existed in the original implementation, so I'm just taking the opportunity to say: `shim.isFunction(handler) === false` would make the line abundantly clear and obviate the need for the comment, even though the comment is already superfluous.

Reply:
yea i just moved this code but I can remove these superfluous comments and fix that conditional